parsec/core/fs/workspacefs/sync_transactions.py

```diff
@@ -224,7 +225,7 @@ def merge_manifests(
     assert remote_version > local_version and local_manifest.need_sync
 
     # All the local changes have been successfully uploaded
-    if local_manifest.match_remote(remote_manifest):
+    if local_manifest.match_remote(remote_manifest):  # type: ignore[arg-type]
```

Comment:
why is this needed ? (and same question for the other type ignore in this file)

Reply:
This is need because both `local_manifest` & `remote_manifest` are unions, but `match_remote` only allow a subset of the union on `remote_manifest`, example:

`LocalFileManifest::match_remote` only take a `FileManifest` not a `FolderManifest` nor an `Union[FileManifest, FolderManifest]`